extensions/amp-story/0.1/amp-story-page.js

```diff
@@ -264,6 +266,32 @@ export class AmpStoryPage extends AMP.BaseElement {
   }
 
 
+  /** @private */
+  toggleAudioIcon_() {
+    // Dispatch event to signal whether audio is playing.
+    const eventType = this.hasAudio_() ?
+        EventType.AUDIO_PLAYING : EventType.AUDIO_STOPPED;
+    dispatch(this.element, eventType, /* opt_bubbles */ true);
+  }
+
+
+  /**
+   * @return {boolean}
+   * @private
+   */
+  hasAudio_() {
+    let hasAudio = false;
+    this.pageElements_.forEach(pageElement => {
+      if (pageElement.hasAudio()) {
+        hasAudio = true;
+        return;
+      }
+    });
+
+    return hasAudio;
```

Comment:
This still executes for every `pageElement` even after `hasAudio` is set. I think it can also be written in a simpler way:

```javascript
hasAudio_() {
  return this.pageElements_.some(pageElement => pageElement.hasAudio());
}
```

Reply:
Done!